### ISSUE TITLE
Card not working in 2023.9

### ISSUE THREAD

**Issue by alexstaal** [pkissling/clock-weather-card#203]
**Checklist:**

- [ x ] I updated to the latest version available
- [ x ] I cleared the cache of my browser
- [ x ] I read the [FAQ](https://github.com/pkissling/clock-weather-card#faq) and the issue I'm having was not mentioned there.

**Release with the issue:**

**Last working release (if known):**
2023.8

**Browser and Operating System:**
Companion app android

**Description of problem:**
Just an empty card, think something to do with the new weather forecast service in 2023.9

**Javascript errors shown in the web inspector (if applicable):**

```

```

**Additional information:**

**Comment by VenimK** [comment]
I'm running 2023.9.1 and it works ok now
Are you using openweathermap ....

**Comment by alexstaal** [comment]
I'm also on 2023.9.1 now, not yet working. I'm using a HACS integration (https://github.com/golles/ha-knmi). However with the standard lovelace weather-forecast card, this is working fine.

**Comment by diplix** [comment]
it’s depending on the weather service you use in the card. if the weather service uses the new weather format, the card is not working, if, like OWM, it uses the old format, the card still works fine.

**Comment by pkissling** [comment]
hi there, thanks for pointing this out. i am on vacation until end of september and didn't even look at the changelog for 2023.9.

**Comment by FL550** [comment]
See here for reference: [Weather attribute](https://www.home-assistant.io/blog/2023/09/06/release-20239/#weather-forecast-service)

**Comment by GerdRuetten** [comment]
> hi there, thanks for pointing this out. i am on vacation until end of september and didn't even look at the changelog for 2023.9 yet. i'm afraid you should not expect a fix before beginning of october

What do you meen with ths reference?

After calling the new service, the card still doesn't work!

**Comment by FL550** [comment]
The service is the new way for retrieving the forecast. The old way ith the attribute is deprecated. As an end user you can't change this and have to wait for the developer to adjust accordingly or use an older version of your weather provider/HA. @GerdRuetten

**Comment by GerdRuetten** [comment]
> The service is the new way for retrieving the forecast. The old way ith the attribute is deprecated. As an end user you can't change this and have to wait for the developer to adjust accordingly or use an older version of your weather provider/HA. @GerdRuetten 

Thank you and sorry!
I misunderstood your post

**Comment by michaelarz** [comment]
While we waiting for a card fix, a small hint for those relying on the integration+card:
creating a template sensor like this (based on example from HA documentation)
```
- trigger:
  - platform: time_pattern
    hours: /1
  action:
    - service: weather.get_forecast
      target:
        entity_id: weather.swis_punkt_swis_punkt
      data:
        type: hourly
      response_variable: hourly
  sensor:
  - name: dwd_forecast
    unique_id: dwd_forecast
    state: "{{ states('sensor.weather_swis_punkt') }}"
    attributes:
      forecast: "{{ hourly.forecast }}"
      temperature: "{{ states('sensor.temperature_swis_punkt') }}"
      temperature_unit: °C
      humidity: "{{ states('sensor.humidity_swis_punkt') }}"
      cloud_coverage: "{{ states('sensor.cloud_coverage_swis_punkt') }}"
      pressure: "{{ states('sensor.pressure_swis_punkt') }}"
      pressure_unit: hPa
      wind_speed: "{{ states('sensor.wind_speed_swis_punkt') }}"
      wind_speed_unit: m/s
      visibility_unit: km
      precipitation_unit: mm
```

 will create an entity having necessary format, which could be then used in the card:
```
  type: custom:clock-weather-card
  entity: sensor.dwd_forecast
```

**Comment by arboeh** [comment]
> While we waiting for a card fix, a small hint for those relying on the integration+card: creating a template sensor like this (based on example from HA documentation)
> 
> ```
> - trigger:
>   - platform: time_pattern
>     hours: /1
>   action:
>     - service: weather.get_forecast
>       target:
>         entity_id: weather.swis_punkt_swis_punkt
>       data:
>         type: hourly
>       response_variable: hourly
>   sensor:
>   - name: dwd_forecast
>     unique_id: dwd_forecast
>     state: "{{ states('sensor.weather_swis_punkt') }}"
>     attributes:
>       forecast: "{{ hourly.forecast }}"
>       temperature: "{{ states('sensor.temperature_swis_punkt') }}"
>       temperature_unit: °C
>       humidity: "{{ states('sensor.humidity_swis_punkt') }}"
>       cloud_coverage: "{{ states('sensor.cloud_coverage_swis_punkt') }}"
>       pressure: "{{ states('sensor.pressure_swis_punkt') }}"
>       pressure_unit: hPa
>       wind_speed: "{{ states('sensor.wind_speed_swis_punkt') }}"
>       wind_speed_unit: m/s
>       visibility_unit: km
>       precipitation_unit: mm
> ```
> 
> will create an entity having necessary format, which could be then used in the card:
> 
> ```
>   type: custom:clock-weather-card
>   entity: sensor.dwd_forecast
> ```

I tried your solution but I definitely do not get the correct values. We have 23°C here right now. I guess it's because the card needs forecast values for every three hours, not hourly.

![Screenshot_20230927-180728~2](https://github.com/pkissling/clock-weather-card/assets/12322114/d7ad4a80-4b55-4f54-bee4-41ec7c32f97c)

Openweathermap does it's job, btw.

**Comment by michaelarz** [comment]
> 
> I tried this "solution" but I definitely do not get the correct values. We have 23°C here right now. I guess it's because the card needs forecast values for every three hours, not hourly.
> 
> ![Screenshot_20230927-180728~2](https://user-images.githubusercontent.com/12322114/271046374-d7ad4a80-4b55-4f54-bee4-41ec7c32f97c.png)
> 
> Openweathermap does it's job, btw.

Mind sharing your template config and state of the sensor?
I had such result with minus temperatures when my sensor had only forecast attribute. After adding those attributes, mentioned in my example it normalised.
Nevertheless, if you can stay with OWM until card got fixed, this seems to be the best "solution". OWM has all necessary
attributes out of the box

**Comment by pkissling** [comment]
just a quick heads up: today i was able to create a working prototype which is able to consume the old _legacy_ weather entity (with `forecast` attribute), as well as the new weather entity (using `get_forecast`). https://github.com/pkissling/clock-weather-card/tree/203-card-not-working-in-20239

the implementation still has some rough edges and some uncovered edge cases. though i'm positive that i will be able to release a new version in the first half of next week.

**Comment by pkissling** [comment]
i think i was able to create a working version. i published it as a beta to get some feedback from you guys.
would you kindly install the 1.1.0 beta and give me some feedback whether the card works for you as well?

<img width="500" alt="image" src="https://github.com/pkissling/clock-weather-card/assets/33731393/d0ded715-5e1b-477b-97f4-3cb1578d8aba">
<img width="500" alt="image" src="https://github.com/pkissling/clock-weather-card/assets/33731393/58bc7f11-9ba7-451f-a2ce-bc851c873cf5">

**Comment by michaelarz** [comment]
Thanks for the fix! Works perfectly fine for me.

**Comment by pkissling** [comment]
I just released `1.1.0`

**Comment by checkiecheck** [comment]
just wanted to check on another weather provider, which seems not to work for me on the beta
https://github.com/golles/ha-knmi
i dont know where to look if card just renders emtpy on this provider
